drivers/display/display_st7789v.c

```diff
@@ -63,33 +63,36 @@ struct st7789v_data {
 static void st7789v_set_lcd_margins(const struct device *dev,
 				    uint16_t x_offset, uint16_t y_offset)
 {
-	struct st7789v_data *data = (struct st7789v_data *)dev->data;
+	struct st7789v_data *data = dev->data;
 
 	data->x_offset = x_offset;
 	data->y_offset = y_offset;
 }
 
 static void st7789v_set_cmd(const struct device *dev, int is_cmd)
 {
-	struct st7789v_config *config = (struct st7789v_config *)dev->config;
+	const struct st7789v_config *config = dev->config;
```

Comment:
commit https://github.com/zephyrproject-rtos/zephyr/pull/36717/commits/6dbb628b7969fcd74fc011672312c7ce54bff809 introduces st7789v_config, therefore this should be fixed there.

Reply:
I squashed these changes and reordered the commits, so that they make more sense.